src/main/java/io/gatling/mojo/GatlingMojo.java

```diff
@@ -37,22 +37,17 @@
 import org.apache.commons.exec.ExecuteException;
 import org.apache.maven.artifact.Artifact;
 import org.apache.maven.artifact.DependencyResolutionRequiredException;
-import org.apache.maven.artifact.resolver.ArtifactResolutionRequest;
-import org.apache.maven.artifact.resolver.ArtifactResolutionResult;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
-import org.apache.maven.plugins.annotations.LifecyclePhase;
-import org.apache.maven.plugins.annotations.Mojo;
-import org.apache.maven.plugins.annotations.Parameter;
-import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.plugins.annotations.*;
 import org.apache.maven.project.MavenProject;
 import org.apache.maven.toolchain.Toolchain;
 import org.codehaus.plexus.util.DirectoryScanner;
 import org.codehaus.plexus.util.ExceptionUtils;
 import org.codehaus.plexus.util.SelectorUtils;
-import org.codehaus.plexus.util.StringUtils;
 
 /** Mojo to execute Gatling. */
+@Execute(phase = LifecyclePhase.TEST_COMPILE)
```

Comment:
I don't understand:
 * Why we need to fork a phase 

     >  Used if your Mojo needs to fork a lifecycle.

    Source: [Execute (annotation)](https://maven.apache.org/plugin-tools/maven-plugin-annotations/apidocs/org/apache/maven/plugins/annotations/Execute.html)

 * Why it's the `TEST_COMPILE` phase.


My understanding is that we should be during the `INTEGRATION_TEST` phase depending on the `TEST_COMPILE`.

Here, I expect issue if the order of plugins declaration is not strict.
(because executions are ordered by phase then, by definition order)

Reply:
> Why we need to fork a phase

Because otherwise, `mvn gatling:test` won't compile the test code and we want to retain compatibility with existing commands.

> Why it's the TEST_COMPILE phase.

That's a good point. I did the same as https://github.com/gatling/gatling-maven-plugin/blob/master/src/main/java/io/gatling/mojo/EnterprisePackageMojo.java#L42

I guess this other one is wrong too and should execute `PACKAGE` instead. cc @tpetillot

Reply:
https://maven.apache.org/developers/mojo-api-specification.html#the-descriptor-and-annotations
> When this goal is invoked, it will first invoke a parallel lifecycle, ending at the given phase. If a goal is provided instead of a phase, that goal will be executed in isolation. The execution of either will not affect the current project, but instead make available the ${executedProject} expression if required. An alternate lifecycle can also be provided: for more information see the documentation on the build lifecycle.

Reply:
We will only recommend triggering the goal. If someone wants to run from maven's lifecycle, we'll expect him to know maven.
What we **need** is that test sources are compiled, not that other tests have run. So `TEST_COMPILE ` looks correct to me. If we end up figuring out it's not, we'll change it.